pkg/kubectl/cmd/create.go

```diff
@@ -122,22 +122,25 @@ func printObjectSpecificMessage(obj runtime.Object, out io.Writer) {
 	switch obj := obj.(type) {
 	case *api.Service:
 		if obj.Spec.Type == api.ServiceTypeLoadBalancer {
-			msg := fmt.Sprintf(`
-			An external load-balanced service was created.  On many platforms (e.g. Google Compute Engine),
-			you will also need to explicitly open a Firewall rule for the service port(s) (%s) to serve traffic.
-
-			See https://github.com/GoogleCloudPlatform/kubernetes/tree/master/docs/services-firewalls.md for more details.
-			`, makePortsString(obj.Spec.Ports, false))
+			msg := fmt.Sprintf(
+				`An external load-balanced service was created.  On some platforms you will
+also need to explicitly open a Firewall rule for the service port(s) (%s) to
+serve traffic.  Some platforms (e.g. GCE and AWS) do this automatically.
+
+See http://releases.k8s.io/HEAD/docs/services-firewalls.md for more details.
+`,
+				makePortsString(obj.Spec.Ports, false))
 			out.Write([]byte(msg))
 		}
 		if obj.Spec.Type == api.ServiceTypeNodePort {
-			msg := fmt.Sprintf(`
-				You have exposed your service on an external port on all nodes in your cluster.
-				If you want to expose this service to the external internet, you may need to set up
-				firewall rules for the service port(s) (%s) to serve traffic.
-				
-				See https://github.com/GoogleCloudPlatform/kubernetes/tree/master/docs/services-firewalls.md for more details.
-				`, makePortsString(obj.Spec.Ports, true))
+			msg := fmt.Sprintf(
+				`You have exposed your service on an external port on all nodes in your
+cluster.  If you want to expose this service to the external internet, you may
+need to set up firewall rules for the service port(s) (%s) to serve traffic.
+
+See http://releases.k8s.io/HEAD/docs/services-firewalls.md for more details.
```

Comment:
Ai gree we could, but we pretty much need that to come from the
cloud-provider, which means from the master.

#11259

On Tue, Jul 14, 2015 at 1:12 PM, Prashanth B <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/create.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11254#discussion_r34613478>
> :
>
> >  			out.Write([]byte(msg))
> >  		}
> >  		if obj.Spec.Type == api.ServiceTypeNodePort {
> > -			msg := fmt.Sprintf(`
> > -				You have exposed your service on an external port on all nodes in your cluster.
> > -				If you want to expose this service to the external internet, you may need to set up
> > -				firewall rules for the service port(s) (%s) to serve traffic.
> > -				
> > -				See https://github.com/GoogleCloudPlatform/kubernetes/tree/master/docs/services-firewalls.md for more details.
> > -				`, makePortsString(obj.Spec.Ports, true))
> > +			msg := fmt.Sprintf(
> > +				`You have exposed your service on an external port on all nodes in your
> > +cluster.  If you want to expose this service to the external internet, you may
> > +need to set up firewall rules for the service port(s) (%s) to serve traffic.
> > +
> > +See http://releases.k8s.io/HEAD/docs/services-firewalls.md for more details.
>
> Nit: since this is kubectl, it feels like we can confirm and tell them one
> way or another (not expecting it in this pr).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11254/files#r34613478>
> .
>